min_avg_waiting_time.py

```diff
@@ -0,0 +1,51 @@
+#!/bin/python3
+# hacker rank problem link: https://www.hackerrank.com/challenges/minimum-average-waiting-time/problem
+
+
+
+from heapq import heappush, heappop
+
+import os
+import sys
+
+#
+# Complete the minimumAverage function below.
+#
+def minimumAverage(customers,n):
+    #
+    # Write your code here.
+    #
+    customers.sort(reverse=True)
+
+    pq = []
+    waiting_time = 0
+    current_time = 0
+
+    while customers or pq:
+        while customers and customers[-1][0] <= current_time:
+            heappush(pq, customers.pop()[::-1])

```

Comment:
It's unclear what algorithm you're following and whether it's a generic algorithm and not a heuristic. I tend to slide with the latter as it doesn't do a thorough check of all possibilities but pick an arbitrary solution.

If this is a well-known solution, it should be named. If not, it's better to stick to a generic algorithm that ensures that we explore the entire problem space to find the optimal solution.